### PR TITLE
ZKWAS-154

### PR DESCRIPTION
**Pull request by rhaoio**
Validate inputs for proof task

### CODE REVIEW

File: src/helper/util.ts [hunk 1]
```diff
@@ -72,65 +72,67 @@ export class ZkWasmUtil {
...
-    let type = inputArray[1];
-    let re1 = new RegExp(/^[0-9A-Fa-f]+$/); // hexdecimal
-    let re2 = new RegExp(/^\d+$/); // decimal
-
-    // Check if value is a number
-    if (!(re1.test(value.slice(2)) || re2.test(value))) {
-      console.log("Error: input value is not an interger number");
-      return null;
+
+  private static validateHex(value: string) {
```
Comment (yymone) [inline src/helper/util.ts:76]: maybe no need to be private? so we can easily test it when use this lib

File: src/helper/util.ts [hunk 2]
```diff
@@ -72,65 +72,67 @@ export class ZkWasmUtil {
...
-        }
-      }
+  // Inputs are strings that should be of the form 32:i64, 0x1234:bytes or 0x1234:bytes-packed and split by spaces
+  static validateInputs(inputs: string): Array<string> {
+    if (inputs === "") {
+      return [];
     }
-    return parsedInputs.flat();
+    // Split the inputs by spaces
+    let inputArray = inputs.split(" ");
```
Comment (yymone) [inline src/helper/util.ts:128]: We better image about some scenario user may input:
1. multi space. For example, input is  "0x01:i64     0x02:i64 0x03:i64", I guess maybe we better accept it? use split(/\s+/)?

File: src/helper/util.ts [hunk 3]
```diff
@@ -72,65 +72,67 @@ export class ZkWasmUtil {
...
-        let args = ZkWasmUtil.parseArg(input);
-        if (args != null) {
-          parsedInputs.push(args);
-        } else {
-          throw Error(`invalid args in ${input}`);
-        }
-      }
+  // Inputs are strings that should be of the form 32:i64, 0x1234:bytes or 0x1234:bytes-packed and split by spaces
+  static validateInputs(inputs: string): Array<string> {
+    if (inputs === "") {
```
Comment (yymone) [inline src/helper/util.ts:124]: if input is pure space? so better trim end space here?

File: src/helper/util.ts (hunk 2 above)
Comment (rhaoio) [reply src/helper/util.ts:128]: it should work with multiple spaces

File: dist/mjs/helper/util.js [hunk 4]
```diff
@@ -53,66 +53,64 @@ export class ZkWasmUtil {
...
-            }
+    // Inputs are strings that should be of the form 32:i64, 0x1234:bytes or 0x1234:bytes-packed and split by spaces
+    static validateInputs(inputs) {
+        let trimmed = inputs.trim();
+        if (trimmed === "") {
+            return [];
         }
-        return parsedInputs.flat();
+        // Split the inputs by spaces
+        let inputArray = trimmed.split(" ");
```
Comment (yymone) [inline dist/mjs/helper/util.js:106]: image
Looks like it cannot handle muptiple space.

**Review by yymone — APPROVED** [review]

**Pull request merged**